Essentials/src/main/java/com/earth2me/essentials/EssentialsUpdateChecker.java

```diff
@@ -0,0 +1,258 @@
+
+    public static boolean isDevBuild() {
+        return devBuild;
+    }
+
+    public static CompletableFuture<UpdateToken> getDevToken() {
+        if (cachedDev == null || ((System.currentTimeMillis() - lastFetchTime) > 1800000L)) {
+            if (pendingDevFuture != null) {
+                return pendingDevFuture;
+            }
+            pendingDevFuture = new CompletableFuture<>();
+            new Thread(() -> {
+                pendingDevFuture.complete(cachedDev = fetchDistance(BRANCH, getVersionIdentifier()));
+                pendingDevFuture = null;
+                lastFetchTime = System.currentTimeMillis();
+            }).start();
+            return pendingDevFuture;
+        }
+        return CompletableFuture.completedFuture(cachedDev);
+    }
+
+    public static CompletableFuture<UpdateToken> getReleaseToken() {
+        if (cachedRelease == null || ((System.currentTimeMillis() - lastFetchTime) > 1800000L)) {
+            if (pendingReleaseFuture != null) {
+                return pendingReleaseFuture;
+            }
+            pendingReleaseFuture = new CompletableFuture<>();
+            new Thread(() -> {
+                catchBlock:
+                try {
+                    final HttpURLConnection connection = (HttpURLConnection) new URL("https://api.github.com/repos/" + REPO + "/releases/latest").openConnection();
+                    connection.connect();
+
+                    if (connection.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+                        // Locally built?
+                        pendingReleaseFuture.complete(cachedRelease = new UpdateToken(BranchStatus.UNKNOWN));
+                        break catchBlock;
+                    }
+                    if (connection.getResponseCode() == HttpURLConnection.HTTP_INTERNAL_ERROR) {
+                        // Github is down
+                        pendingReleaseFuture.complete(new UpdateToken(BranchStatus.ERROR));
+                        break catchBlock;
+                    }
+
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream(), Charsets.UTF_8))) {
+                        latestRelease = new Gson().fromJson(reader, JsonObject.class).get("tag_name").getAsString();
+                        pendingReleaseFuture.complete(cachedRelease = fetchDistance(latestRelease, getVersionIdentifier()));
+                    } catch (JsonSyntaxException | NumberFormatException e) {
+                        e.printStackTrace();
+                        pendingReleaseFuture.complete(new UpdateToken(BranchStatus.ERROR));
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    pendingReleaseFuture.complete(new UpdateToken(BranchStatus.ERROR));
+                }
+                pendingReleaseFuture = null;
+                lastFetchTime = System.currentTimeMillis();
+            }).start();
+        }
```

Comment:
```suggestion
        return pendingReleaseFuture;
        }
```
Without this, we would just return the cached result. On first run this is null and causes NPEs:

![error](https://i.imgur.com/mM43mqW.png)